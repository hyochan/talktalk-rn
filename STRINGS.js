import LocalizedStrings from 'react-native-localization';

export const strings = new LocalizedStrings({
  en: {
    LOADING: 'LOADING...',
    HELLO: 'Hello',
    LOGIN: 'Login',
    EMAIL: 'Email',
    PASSWORD: 'Password',
    SIGN_UP: 'Sign Up',
    LOGIN: 'Login',
    LOGOUT: 'Logout',
    FORGOT_PW: 'Forgot password?',
    FIND_PW: 'Find Password',
    NAME: 'name',
    STATUS_MSG: 'Status message',
    REGISTER: 'Register',
    SEND_LINK: 'Send Link',
    FRIEND: 'Friend',
    MESSAGE: 'Message',
    MY_PROFILE: 'My profile',
    UPDATE: 'Update',
    NO_CONTENT: 'No content',
    ADD_FRIEND: 'ADD',
    DELETE_FRIEND: 'DELETE',
    SEARCH_USER: 'Search User',
    CHAT: 'CHAT',
    SEND: 'Send',
    WRITE_MESSAGE: 'Write message...',
    ERROR: 'Error',
    FRIEND_ADDED: 'Added to your friend.',
    FRIEND_ALREADY_ADDED: 'Already your friend.',
  },
  ko: {
    LOADING: '로딩중...',
    HELLO: '안녕하세요',
    LOGIN: '로그인',
    LOGOUT: '로그아웃',
    EMAIL: '이메일',
    PASSWORD: '패스워드',
    SIGN_UP: '회원가입',
    LOGIN: '로그인',
    FORGOT_PW: '비밀번호를 잊어버리셨나요?',
    FIND_PW: '비밀번호 찾기',
    NAME: '이름',
    STATUS_MSG: '상태메세지',
    REGISTER: '가입',
    SEND_LINK: '링크 보내기',
    FRIEND: '친구',
    MESSAGE: '메세지',
    MY_PROFILE: '나의 프로필',
    UPDATE: '수정',
    NO_CONTENT: '컨텐츠가 없습니다',
    ADD_FRIEND: '친구추가',
    DELETE_FRIEND: '친구삭제',
    SEARCH_USER: '사용자 검색',
    CHAT: '채팅',
    SEND: '보내기',
    WRITE_MESSAGE: '메세지를 입력해주세요…',
    Error: '에러',
    FRIEND_ADDED: '친구로 등록되었습니다.',
    FRIEND_ALREADY_ADDED: '이미 친구입니다.',
  },
});

export const getString = (str) => {
  return strings[str];
};
