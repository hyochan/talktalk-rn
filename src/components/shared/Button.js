// @flow
import React, { Component } from 'react';
import {
  ActivityIndicator,
  StyleSheet,
  TouchableOpacity,
  Image,
  Text,
  View,
} from 'react-native';
import styled from 'styled-components/native';

import { ratio, colors } from '../../utils/Styles';

const StyledContainer = styled.View`
  flex: 1;
`;

const StyledButton = styled.View`
  background-color: ${colors.dodgerBlue};
  border-color: ${colors.dodgerBlue};
  border-radius: 4;
  border-width: 1;
  shadow-color: ${colors.dodgerBlue};
  shadow-radius: 4;
  shadow-opacity: 0.3;
  align-items: center;
  justify-content: center;
`;

const StyledButtonWhite = styled(StyledButton)`
  background-color: white;
  shadow-radius: 0;
  shadow-opacity: 0.0;
`;

const StyledButtonDisabled = styled.View`
  background-color: rgb(243,243,243);
  align-self: center;
  border-radius: 4;
  border-width: 2;
  border-color: #333;
  align-items: center;
  justify-content: center;
`;

const StyledText = styled.Text`
  font-size: 14;
  font-weight: bold;
  color: ${colors.dodgerBlue};
`;

const StyledTextWhite = styled(StyledText)`
  color: white;
`;

const StyledImageLeft = styled.Image`
  width: 24;
  height: 24;
  position: absolute;
  left: 16;
`;

type Props = {
  isWhite?: boolean;
  isLoading?: boolean;
  isDisabled?: boolean;
  onPress?: () => void;
  imgLeftSrc?: any;
  indicatorColor?: string;
  activeOpacity?: number;
  children?: any;
  width?: number;
  height?: number;
};

type State = {

};

class Button extends Component<Props, State> {
  static defaultProps: Props = {
    isWhite: false,
    isLoading: false,
    isDisabled: false,
    indicatorColor: 'white',
    activeOpacity: 0.5,
    width: 136,
    height: 60,
  };

  render() {
    if (this.props.isDisabled) {
      return (
        <StyledContainer>
          <StyledButtonDisabled>
            <StyledText>{this.props.children}</StyledText>
          </StyledButtonDisabled>
        </StyledContainer>
      );
    }
    if (this.props.isLoading) {
      return (
        <StyledContainer>
          <StyledButton>
            <ActivityIndicator size='small' color={this.props.indicatorColor} />
          </StyledButton>
        </StyledContainer>
      );
    }
    return (
      <StyledContainer>
        <TouchableOpacity
          activeOpacity={this.props.activeOpacity}
          onPress={this.props.onPress}
        >
          {
            !this.props.isWhite
              ? <StyledButton>
                {this.renderContent()}
              </StyledButton>
              : <StyledButtonWhite>
                {this.renderContent()}
              </StyledButtonWhite>
          }
        </TouchableOpacity>
      </StyledContainer>
    );
  }

  renderContent = () => {
    return <View style={{
      width: this.props.width,
      height: this.props.height,
      justifyContent: 'center',
      alignItems: 'center',
    }}>
      {
        this.props.imgLeftSrc
          ? <StyledImageLeft
            source={this.props.imgLeftSrc}
          />
          : null
      }
      { this.renderText() }
    </View>;
  }

  renderText = () => {
    return (
      this.props.isWhite
        ? <StyledText>{this.props.children}</StyledText>
        : <StyledTextWhite>{this.props.children}</StyledTextWhite>
    );
  }
}

export default Button;
