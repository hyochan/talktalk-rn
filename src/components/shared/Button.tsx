import React from 'react';
import {
  ActivityIndicator,
  StyleSheet,
  TouchableOpacity,
  ViewStyle,
  Image,
  Text,
  View,
} from 'react-native';
import styled from 'styled-components/native';

interface IStyledElement {
  white?: boolean;
};

interface IButtonContainer {
  width: number;
  height: number;
};

interface IProps {
  testID: string;
  containerStyle: ViewStyle;
  isWhite: boolean;
  isLoading: boolean;
  isDisabled: boolean;
  onPress: () => void;
  imgLeftSrc: any;
  indicatorColor: string;
  activeOpacity: number;
  children: any;
  width: number;
  height: number;
}

const StyledContainer = styled.View`
  flex: 1;
`;

const StyledButtonContainer = styled.View<IButtonContainer>`
  width: ${({ width }) => `${width}px`};
  height: ${({ height }) => `${height}px`};
  justify-content: center;
  align-items: center;
`;

const StyledButton = styled.View<IStyledElement>`
  background-color: ${({ white, theme: { colors: { dodgerBlue }}}) => white ? 'white' : dodgerBlue};
  border-color: ${({ theme: { colors: { dodgerBlue }}}) => dodgerBlue};
  border-radius: 4;
  border-width: 1;
  shadow-color: ${({ theme: { colors: { dodgerBlue }}}) => dodgerBlue};
  shadow-radius: ${({ white }) => white ? 0 : 4};
  shadow-opacity: ${({ white }) => white ? 0.0 : 0.3};
  align-items: center;
  justify-content: center;
`;

const StyledButtonDisabled = styled.View`
  background-color: rgb(243,243,243);
  align-self: center;
  border-radius: 4;
  border-width: 2;
  border-color: #333;
  align-items: center;
  justify-content: center;
`;

const StyledText = styled.Text<IStyledElement>`
  font-size: 14;
  font-weight: bold;
  color: ${({ white, theme: { colors: { dodgerBlue }}}) => white ? 'white' : dodgerBlue};
`;

const StyledImageLeft = styled.Image`
  width: 24;
  height: 24;
  position: absolute;
  left: 16;
`;

function Button({
  isDisabled,
  isWhite,
  activeOpacity,
  onPress,
  isLoading,
  width,
  height,
  indicatorColor,
  imgLeftSrc,
  children
}: Partial<IProps>) { 
  return (
    <StyledContainer>
      {isDisabled ? (
        <StyledButtonDisabled>
          <StyledText>{children}</StyledText>
        </StyledButtonDisabled>
      ) : (
        <TouchableOpacity
          testID="press_id"
          activeOpacity={activeOpacity}
          onPress={onPress}
        >
          <StyledButton white={isWhite}>
            <StyledButtonContainer
              width={width}
              height={height}
            >
              {isLoading
                ? <ActivityIndicator size='small' color={indicatorColor} />
                : (imgLeftSrc ? <StyledImageLeft source={imgLeftSrc} /> : undefined)
              }
              {!isLoading
                ? <StyledText white={!isWhite}>{children}</StyledText>
                : undefined
              }
            </StyledButtonContainer>
          </StyledButton>            
        </TouchableOpacity>
      )}      
    </StyledContainer>
  );
}

Button.defaultProps = {
  isWhite: false,
  isLoading: false,
  isDisabled: false,
  indicatorColor: 'white',
  activeOpacity: 0.5,
  width: 136,
  height: 60,
};

export default Button;
