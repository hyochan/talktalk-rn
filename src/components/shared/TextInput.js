// @flow
import React, { useState } from 'react';
import {
  ActivityIndicator,
  StyleSheet,
  Image,
  Text,
  View,
  TextInput,
} from 'react-native';

import type {
  ____ViewStyleProp_Internal as ViewStyle,
  ____TextStyleProp_Internal as TextStyle,
  ____ImageStyleProp_Internal as ImageStyle,
} from 'react-native/Libraries/StyleSheet/StyleSheetTypes';

import { ratio, colors } from '../../utils/Styles';

type Styles = {
  wrapper: ViewStyle,
  label: ViewStyle,
  labelFocus: TextStyle,
  input: ViewStyle,
  inputFocus: TextStyle,
};

const styles: Styles = StyleSheet.create({
  wrapper: {
    alignSelf: 'stretch',

    flexDirection: 'column',
    justifyContent: 'flex-start',
    alignItems: 'flex-start',
  },
  label: {
    color: colors.blueyGray,
    marginBottom: 8,
    fontSize: 12,
  },
  labelFocus: {
    color: colors.dodgerBlue,
    marginBottom: 8,
    fontSize: 12,
  },
  input: {
    alignSelf: 'stretch',
    color: colors.dusk,
    fontSize: 16,
    paddingVertical: 22,
    paddingHorizontal: 20,
    borderWidth: 1,
    borderColor: 'rgb(233,237,244)',

    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'flex-start',
  },
  inputFocus: {
    alignSelf: 'stretch',
    color: colors.dusk,
    fontSize: 16,
    paddingVertical: 22,
    paddingHorizontal: 20,
    borderWidth: 1,
    borderColor: colors.dodgerBlue,

    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'flex-start',
  },
});

type Props = {
  style?: ViewStyle;
  labelStyle?: TextStyle;
  labelStyleFocus?: TextStyle;
  placeholderTextColor?: string;
  inputStyle?: TextStyle;
  isPassword?: boolean;
  multiline?: boolean;
  txtLabel?: string;
  txtHint?: string;
  txt?: string;
  onFocus?: () => void;
  onTextChanged?: (text: string) => void;
  onSubmitEditing?: (text: string) => void;
  returnKeyType?: any;
};

type State = {
  focused: boolean;
};

function Shared(props: Props, state: State) {
  const [focused, setFocused] = useState(false);

  const renderTxtLabel = () => {
    if (props.txtLabel) {
      return (
        <Text style={focused ? props.labelStyleFocus : props.labelStyle}>
          {props.txtLabel}
        </Text>
      );
    }
    return null;
  };

  return (
    <View style={[
      styles.wrapper,
      props.style,
    ]}>
      {renderTxtLabel()}
      <TextInput
        style={[
          focused ? styles.inputFocus : styles.input,
          props.inputStyle,
        ]}
        underlineColorAndroid='transparent' // android fix
        autoCapitalize='none'
        autoCorrect={false}
        multiline={props.multiline}
        onChangeText={props.onTextChanged}
        value={props.txt}
        onFocus={() => setFocused(true)}
        onBlur={() => setFocused(false)}
        placeholder={props.txtHint}
        placeholderTextColor={props.placeholderTextColor}
        // onSubmitEditing={this.props.onSubmitEditing}
        returnKeyType={props.returnKeyType}
        secureTextEntry={props.isPassword}
      />
    </View>
  );
}

Shared.defaultProps = {
  style: styles.wrapper,
  labelStyle: styles.label,
  labelStyleFocus: styles.labelFocus,
};

export default Shared;
