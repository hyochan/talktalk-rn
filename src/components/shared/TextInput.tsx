import React, { useState } from 'react';
import {
  ActivityIndicator,
  StyleSheet,
  Image,
  Text,
  View,
  TextInput,
  ViewStyle,
  TextStyle,
} from 'react-native';

import { ratio, colors } from '../../utils/Styles';

interface IStyles {
  wrapper: ViewStyle;
  label: ViewStyle;
  labelFocus: TextStyle;
  input: ViewStyle;
  inputFocus: TextStyle;
}

const styles: IStyles = StyleSheet.create({
  wrapper: {
    alignSelf: 'stretch',

    flexDirection: 'column',
    justifyContent: 'flex-start',
    alignItems: 'flex-start',
  },
  label: {
    color: colors.blueyGray,
    marginBottom: 8,
    fontSize: 12,
  },
  labelFocus: {
    color: colors.dodgerBlue,
    marginBottom: 8,
    fontSize: 12,
  },
  input: {
    alignSelf: 'stretch',
    color: colors.dusk,
    fontSize: 16,
    paddingVertical: 22,
    paddingHorizontal: 20,
    borderWidth: 1,
    borderColor: 'rgb(233,237,244)',

    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'flex-start',
  },
  inputFocus: {
    alignSelf: 'stretch',
    color: colors.dusk,
    fontSize: 16,
    paddingVertical: 22,
    paddingHorizontal: 20,
    borderWidth: 1,
    borderColor: colors.dodgerBlue,

    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'flex-start',
  },
});

interface IProps {
  testID?: string;
  style?: ViewStyle;
  labelStyle?: TextStyle;
  labelStyleFocus?: TextStyle;
  placeholderTextColor?: string;
  inputStyle?: TextStyle;
  isPassword?: boolean;
  multiline?: boolean;
  txtLabel?: string;
  txtHint?: string;
  txt?: string;
  onFocus?: () => void;
  onTextChanged?: (text: string) => void;
  onSubmitEditing?: (text: string) => void;
  returnKeyType?: any;
}

interface IState {
  focused: boolean;
}

function Shared(props: IProps, state: IState) {
  const [focused, setFocused] = useState(false);

  const renderTxtLabel = () => {
    if (props.txtLabel) {
      return (
        <Text style={focused ? props.labelStyleFocus : props.labelStyle}>
          {props.txtLabel}
        </Text>
      );
    }
    return null;
  };

  return (
    <View style={[
      styles.wrapper,
      props.style,
    ]}>
      {renderTxtLabel()}
      <TextInput
        style={[
          focused ? styles.inputFocus : styles.input,
          props.inputStyle,
        ]}
        underlineColorAndroid='transparent' // android fix
        autoCapitalize='none'
        autoCorrect={false}
        multiline={props.multiline}
        onChangeText={props.onTextChanged}
        value={props.txt}
        onFocus={() => setFocused(true)}
        onBlur={() => setFocused(false)}
        placeholder={props.txtHint}
        placeholderTextColor={props.placeholderTextColor}
        // onSubmitEditing={this.props.onSubmitEditing}
        returnKeyType={props.returnKeyType}
        secureTextEntry={props.isPassword}
      />
    </View>
  );
}

Shared.defaultProps = {
  style: styles.wrapper,
  labelStyle: styles.label,
  labelStyleFocus: styles.labelFocus,
};

export default Shared;
