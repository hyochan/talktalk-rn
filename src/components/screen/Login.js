// @flow
import React, { Component } from 'react';
import {
  StyleSheet,
  TouchableOpacity,
  Image,
  Text,
  View,
  SafeAreaView,
  ScrollView,
} from 'react-native';

import type {
  ____ViewStyleProp_Internal as ViewStyle,
  ____TextStyleProp_Internal as TextStyle,
  ____ImageStyleProp_Internal as ImageStyle,
} from 'react-native/Libraries/StyleSheet/StyleSheetTypes';

import TextInput from '../shared/TextInput';
import Button from '../shared/Button';
import StatusBar from '../shared/StatusBar';

import { IC_ICON } from '../../utils/Icons';
import { statusBarHeight, ratio, colors } from '../../utils/Styles';
import { getString } from '../../../STRINGS';

import styled from 'styled-components/native';

const StyledScollView = styled.ScrollView`
  background-color: white;
`;

const StyledContainer = styled.View`
  flex: 1;
  flex-direction: column;
  align-items: center;
`;

const StyledIconWrapper = styled.View`
  position: absolute;
  top: 76;
  left: 40;

  flex-direction: column;
  align-items: flex-start;
`;

const StyledIcon = styled.Image`
  width: 60;
  height: 60;
`;

const StyledIconText = styled.Text`
  color: ${colors.dusk},
  font-size: 20;
  font-weight: bold;
  margin-top: 8;
`;

const StyledInputWrapper = styled.View`
  margin-top: 260;
  width: 78%;
  height: 300;

  flex-direction: column;
  align-items: center;
`;

const StyledButtonWrapper = styled.View`
  align-self: stretch;
  margin-top: 20;
  height: 60;

  flex-direction: row;
  justify-content: space-between;
  align-items: center;
`;

const StyledButtonSignup = styled.View`
  background-color: transparent;
  border-radisu: 4;
  border-width: 1;
  height: 100%;
  width: 100%;

  border-color: ${colors.dodgerBlue};
  align-items: center;
  justify-content: center;
`;

const StyledTextSignup = styled.Text`
  font-size: 16;
  font-weight: bold;
  color: ${colors.dodgerBlue};
`;

const StyledButtonLogin = styled.View`
  background-color: ${colors.dodgerBlue};
  border-color: ${colors.dodgerBlue};
  border-radius: 4;
  border-width: 1;
  height: 60;
  shadow-color: ${colors.dodgerBlue};
  showdow-offset: {
    width: 0;
    height: 10;
  };
  shadow-radius: 4;
  shadow-opacity: 0.3;
  align-items: center;
  justify-content: center;
`;

const StyledTextLogin = styled.TextInput`
  font-size: 16;
  font-weight: bold;
  color: white;
`;

const StyledTextForgotPw = styled.Text`
  font-size: 12;
  color: ${colors.dodgerBlue};
  text-decoration-line: underline;
`;

const StyledTextCopyright = styled.Text`
  margin-top: 80;
  font-size: 12;
  color: ${colors.cloudyBlue};
`;

type Props = {
  navigation: any;
};
type State = {
  isLoggingIn: boolean;
  email: string;
  pw: string;
};

class Screen extends Component<Props, State> {
  static navigationOptions = {
    headerStyle: {
      borderBottomWidth: 0,
    }
  };

  constructor(props: Props) {
    super(props);
    this.state = {
      isLoggingIn: false,
      email: '',
      pw: '',
    };
  }

  render() {
    return (
      <SafeAreaView>
        <StyledScollView>
          <StatusBar isDarkContent={true}/>
          <StyledContainer>
            <StyledIconWrapper>
              <StyledIcon source={IC_ICON}/>
              <StyledIconText>{getString('HELLO')}.</StyledIconText>
            </StyledIconWrapper>
            <StyledInputWrapper>
              <TextInput
                // txtLabel={ getString('EMAIL') }
                txtHint={ getString('EMAIL') }
                txt={ this.state.email }
                onTextChanged={ (text) => this.onTextChanged('EMAIL', text)}
              />
              <TextInput
                style={{ marginTop: 8 }}
                // txtLabel={ getString('EMAIL') }
                txtHint={ getString('PASSWORD') }
                txt={ this.state.pw }
                onTextChanged={ (text) => this.onTextChanged('PW', text)}
                isPassword={ true }
              />
              <StyledButtonWrapper>
                <Button
                  id='signup'
                  constainerStyle={{ flex: 1 }}
                  onPress={() => this.goToSignup()}
                  isWhite
                >{getString('SIGN_UP')}</Button>
                <View style={{ width: 8 }}/>
                <Button
                  id='login'
                  constainerStyle={{ flex: 1 }}
                  isLoading={this.state.isLoggingIn}
                  onPress={() => this.onLogin()}
                >{getString('LOGIN')}</Button>
              </StyledButtonWrapper>
              <View style={{ height: 16 }} />
              <TouchableOpacity
                activeOpacity={0.5}
                onPress={() => this.goToFindPw()}
              >
                <StyledTextForgotPw>{getString('FORGOT_PW')}</StyledTextForgotPw>
              </TouchableOpacity>
              <StyledTextCopyright>copyright by dooboolab.com</StyledTextCopyright>
            </StyledInputWrapper>
          </StyledContainer>
        </StyledScollView>
      </SafeAreaView>
    );
  }

  onTextChanged = (type: string, text: string) => {
    switch (type) {
      case 'EMAIL':
        this.setState({ email: text });
        return;
      case 'PW':
        this.setState({ pw: text });
    }
  }

  goToSignup = () => {
    this.props.navigation.navigate('Signup');
    // this.props.navigation.navigate('Signup');
  }
  goToFindPw = () => {
    this.props.navigation.navigate('FindPw');
  }

  onLogin = () => {
    this.setState({ isLoggingIn: true }, () => {
      try {
      } catch (err) {
        // Alert.alert(getString('ERROR'), err.message);
      } finally {
        this.setState({ isLoggingIn: false });
      }
    });
  }
}

export default Screen;
