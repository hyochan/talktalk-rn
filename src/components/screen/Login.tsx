import React, { Component, useState } from 'react';
import {
  StyleSheet,
  TouchableOpacity,
  Image,
  Text,
  View,
  SafeAreaView,
  ScrollView,
} from 'react-native';
import { NavigationScreenProp, NavigationStateRoute } from 'react-navigation';

import TextInput from '../shared/TextInput';
import Button from '../shared/Button';
import StatusBar from '../shared/StatusBar';

import { IC_ICON } from '../../utils/Icons';
import { statusBarHeight, ratio, colors } from '../../utils/Styles';
import { getString } from '../../../STRINGS';

import styled from 'styled-components/native';

const StyledScollView = styled.ScrollView`
  background-color: white;
`;

const StyledContainer = styled.View`
  flex: 1;
  flex-direction: column;
  align-items: center;
`;

const StyledIconWrapper = styled.View`
  position: absolute;
  top: 76;
  left: 40;

  flex-direction: column;
  align-items: flex-start;
`;

const StyledIcon = styled.Image`
  width: 60;
  height: 60;
`;

const StyledIconText = styled.Text`
  color: ${colors.dusk},
  font-size: 20;
  font-weight: bold;
  margin-top: 8;
`;

const StyledInputWrapper = styled.View`
  margin-top: 260;
  width: 78%;
  height: 300;

  flex-direction: column;
  align-items: center;
`;

const StyledButtonWrapper = styled.View`
  align-self: stretch;
  margin-top: 20;
  height: 60;

  flex-direction: row;
  justify-content: space-between;
  align-items: center;
`;

const StyledButtonSignUp = styled.View`
  background-color: transparent;
  border-radisu: 4;
  border-width: 1;
  height: 100%;
  width: 100%;

  border-color: ${colors.dodgerBlue};
  align-items: center;
  justify-content: center;
`;

const StyledTextSignUp = styled.Text`
  font-size: 16;
  font-weight: bold;
  color: ${colors.dodgerBlue};
`;

const StyledButtonLogin = styled.View`
  background-color: ${colors.dodgerBlue};
  border-color: ${colors.dodgerBlue};
  border-radius: 4;
  border-width: 1;
  height: 60;
  shadow-color: ${colors.dodgerBlue};
  showdow-offset: {
    width: 0;
    height: 10;
  };
  shadow-radius: 4;
  shadow-opacity: 0.3;
  align-items: center;
  justify-content: center;
`;

const StyledTextLogin = styled.TextInput`
  font-size: 16;
  font-weight: bold;
  color: white;
`;

const StyledTextForgotPw = styled.Text`
  font-size: 12;
  color: ${colors.dodgerBlue};
  text-decoration-line: underline;
`;

const StyledTextCopyright = styled.Text`
  margin-top: 80;
  font-size: 12;
  color: ${colors.cloudyBlue};
`;

interface IProps {
  navigation: any;
}

interface IState {
  isLoggingIn: boolean;
  email: string;
  pw: string;
}

function Screen(props: IProps, state: IState) {
  const [isLoggingIn, setIsLoggingIn] = useState(false);
  const [email, setEmail] = useState('');
  const [pw, setPw] = useState('');
  let timer: any;

  const onTextChanged = (type: string, text: string) => {
    switch (type) {
      case 'EMAIL':
        setEmail(text);
        break;
      case 'PW':
        setPw(text);
        break;
    }
  };

  const goToSignUp = () => {
    props.navigation.navigate('SignUp');
    // props.navigation.navigate('SignUp');
  };

  const goToFindPw = () => {
    props.navigation.navigate('FindPw');
  };

  const onLogin = () => {
    setIsLoggingIn(true);
    timer = setTimeout(() => {
      setIsLoggingIn(false);
      clearTimeout(timer);
    }, 1000);
  };

  return (
    <SafeAreaView>
      <StyledScollView>
        <StatusBar isDarkContent={true}/>
        <StyledContainer>
          <StyledIconWrapper>
            <StyledIcon source={IC_ICON}/>
            <StyledIconText>{getString('HELLO')}.</StyledIconText>
          </StyledIconWrapper>
          <StyledInputWrapper>
            <TextInput
              testID='email_input'
              // txtLabel={ getString('EMAIL') }
              txtHint={ getString('EMAIL') }
              txt={ email }
              onTextChanged={ (text) => onTextChanged('EMAIL', text)}
            />
            <TextInput
              testID='pw_input'
              style={{ marginTop: 8 }}
              // txtLabel={ getString('EMAIL') }
              txtHint={ getString('PASSWORD') }
              txt={ pw }
              onTextChanged={ (text) => onTextChanged('PW', text)}
              isPassword={ true }
            />
            <StyledButtonWrapper>
              <Button
                testID='btnSignUp'
                containerStyle={{ flex: 1 }}
                onPress={() => goToSignUp()}
                isWhite
              >{getString('SIGN_UP')}</Button>
              <View style={{ width: 8 }}/>
              <Button
                testID='btnLogin'
                containerStyle={{ flex: 1 }}
                isLoading={isLoggingIn}
                onPress={() => onLogin()}
              >{getString('LOGIN')}</Button>
            </StyledButtonWrapper>
            <View style={{ height: 16 }} />
            <TouchableOpacity
              testID='findPw'
              activeOpacity={0.5}
              onPress={() => goToFindPw()}
            >
              <StyledTextForgotPw>{getString('FORGOT_PW')}</StyledTextForgotPw>
            </TouchableOpacity>
            <StyledTextCopyright>copyright by dooboolab.com</StyledTextCopyright>
          </StyledInputWrapper>
        </StyledContainer>
      </StyledScollView>
    </SafeAreaView>
  );
}

export default Screen;
