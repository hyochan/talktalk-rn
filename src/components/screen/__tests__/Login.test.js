import 'react-native';
import * as React from 'react';
import Login, { styles } from '../Login';
import  { getString } from '../../../../STRINGS';
import TextInput from '../../shared/TextInput';
import Button from '../../shared/Button';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';
import { shallow, render } from 'enzyme';

let props;
let wrapper;
let tree;
let instance;

describe('rendering test', () => {
  beforeEach(() => {
    props = {
      navigation: {
        navigate: jest.fn(),
      },
    };
    wrapper = shallow(<Login {...props} />);
    tree = renderer.create(<Login {...props} />).toJSON();
    instance = renderer.create(<Login {...props} />).getInstance();
  });
  const findElement = function (tree, element) {
    let result = undefined;
    for (let node in tree.children) {
      if (tree.children[node].props.txt==element) {
        result = true;
      }
    }
    return result;
  };

  it('renders as expected', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('renders initial state', () => {
    expect(instance.state.email).toEqual('');
    expect(instance.state.pw).toEqual('');
    expect(instance.state.isLoggingIn).toEqual(false);
  });

  it('find all components', () => {
    // expect(wrapper.find('StyledScollView')).toHaveLength(1);
    // expect(wrapper.find('StatusBar')).toHaveLength(0);
    expect(wrapper.find(TextInput)).toHaveLength(2);
    expect(wrapper.find(Button)).toHaveLength(2);
    // expect(wrapper.find('Text').at(0)).toHaveLength(1);
    // expect(wrapper.find('Text').at(0).children).toHaveLength(1);
  });

  it('find elements', () => {
    expect(findElement(tree, getString('EMAIL')));
    expect(findElement(tree, getString('PASSWORD')));
  });

  it('all functions defined', () => {
    expect(instance.onLogin).toBeDefined;
    expect(instance.onTextChanged).toBeDefined;
    expect(instance.goToSignup).toBeDefined;
    expect(instance.goToForgotPw).toBeDefined;
  });

  it('onTextChanged', () => {
    instance.onTextChanged('EMAIL', 'aa@aa.aa');
    expect(instance.state.email).toEqual('aa@aa.aa');
    instance.onTextChanged('PW', 'aaaaaa');
    expect(instance.state.pw).toEqual('aaaaaa');
  });
});

describe('interaction', () => {
  beforeEach(() => {
    props = {
      navigation: {
        navigate: jest.fn(),
      },
    };
    wrapper = shallow(<Login {...props} />);
    tree = renderer.create(<Login {...props} />).toJSON();
    instance = renderer.create(<Login {...props} />).getInstance();
  });
  describe('clicking the button', () => {
    it('should call onLogin callback', () => {
      const spy = jest.spyOn(wrapper.instance(), 'onLogin');
      const loginBtn = wrapper.find('#login');
      loginBtn.props().onPress();
      expect(spy).toHaveBeenCalled();
      expect(wrapper.instance().state.isLoggingIn).toEqual(false);
      // wrapper.update();
      // expect(wrapper.instance().state.isLoggingIn).toEqual(true);
      // wrapper.update();
      // expect(wrapper.instance().state.isLoggingIn).toEqual(false);
    });
    it('should call goToSignup callback', () => {
      const spy = jest.spyOn(wrapper.instance(), 'goToSignup');
      const signupBtn = wrapper.find('#signup');
      signupBtn.props().onPress();
      expect(spy).toBeCalled();
      expect(props.navigation.navigate).toHaveBeenCalledWith('Signup');
    });
    it('should call goToForgotPw callack', () => {
      const findPwBtn = wrapper.find('TouchableOpacity');
      findPwBtn.props().onPress();
      expect(props.navigation.navigate).toHaveBeenCalledWith('FindPw');
    });
  });
  afterAll(() => {
    wrapper.instance().prototype.onLogin.mockRestore();
    wrapper.instance().prototype.goToSignup.mockRestore();
    wrapper.instance().prototype.goToFindPw.mockRestore();
  });
});