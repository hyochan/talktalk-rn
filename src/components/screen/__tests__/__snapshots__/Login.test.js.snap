// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rendering test renders as expected 1`] = `
<SafeAreaView>
  <Styled(ScrollView)>
    <Shared
      isDarkContent={true}
    />
    <Styled(View)>
      <Styled(View)>
        <Styled(Component)
          source={
            Object {
              "testUri": "../../../assets/icons/icon.png",
            }
          }
        />
        <Styled(Text)>
          Hello
          .
        </Styled(Text)>
      </Styled(View)>
      <Styled(View)>
        <Shared
          labelStyle={
            Object {
              "color": "rgb(134,154,183)",
              "fontSize": 12,
              "marginBottom": 8,
            }
          }
          labelStyleFocus={
            Object {
              "color": "rgb(58,139,255)",
              "fontSize": 12,
              "marginBottom": 8,
            }
          }
          onTextChanged={[Function]}
          style={
            Object {
              "alignItems": "flex-start",
              "alignSelf": "stretch",
              "flexDirection": "column",
              "justifyContent": "flex-start",
            }
          }
          txt=""
          txtHint="Email"
        />
        <Shared
          isPassword={true}
          labelStyle={
            Object {
              "color": "rgb(134,154,183)",
              "fontSize": 12,
              "marginBottom": 8,
            }
          }
          labelStyleFocus={
            Object {
              "color": "rgb(58,139,255)",
              "fontSize": 12,
              "marginBottom": 8,
            }
          }
          onTextChanged={[Function]}
          style={
            Object {
              "marginTop": 8,
            }
          }
          txt=""
          txtHint="Password"
        />
        <Styled(View)>
          <Button
            activeOpacity={0.5}
            constainerStyle={
              Object {
                "flex": 1,
              }
            }
            height={60}
            id="signup"
            indicatorColor="white"
            isDisabled={false}
            isLoading={false}
            isWhite={true}
            onPress={[Function]}
            width={136}
          >
            Sign Up
          </Button>
          <View
            style={
              Object {
                "width": 8,
              }
            }
          />
          <Button
            activeOpacity={0.5}
            constainerStyle={
              Object {
                "flex": 1,
              }
            }
            height={60}
            id="login"
            indicatorColor="white"
            isDisabled={false}
            isLoading={false}
            isWhite={false}
            onPress={[Function]}
            width={136}
          >
            Login
          </Button>
        </Styled(View)>
        <View
          style={
            Object {
              "height": 16,
            }
          }
        />
        <TouchableOpacity
          activeOpacity={0.5}
          onPress={[Function]}
        >
          <Styled(Text)>
            Forgot password?
          </Styled(Text)>
        </TouchableOpacity>
        <Styled(Text)>
          copyright by dooboolab.com
        </Styled(Text)>
      </Styled(View)>
    </Styled(View)>
  </Styled(ScrollView)>
</SafeAreaView>
`;
